packages/office-ui-fabric-react/src/components/Callout/CalloutContent.tsx

```diff
@@ -79,7 +79,8 @@ export class CalloutContent extends BaseComponent<ICalloutProps, ICalloutState>
   }
 
   public componentWillUpdate(newProps: ICalloutProps) {
-    if (newProps.targetElement !== this.props.targetElement || newProps.target !== this.props.target) {
+    // If the target element changed, find the new one. If we are tracking target with class name, always find element because we do not know if fabric has rendered a new element and disposed the old element.
+    if (newProps.targetElement !== this.props.targetElement || newProps.target !== this.props.target || typeof(newProps.target) === 'string' || newProps.target instanceof String) {
```

Comment:
@joschect mind taking another look here? I think this is probably ok, but i haven't tested or thought through the implications of remeasuring on every componentWillUpdate? This seems like it could cause extraneous reflows in the browser, if we're clearing the height every time when using a string based target.